app/questionnaire_state/state_item.py

```diff
@@ -40,6 +44,23 @@ def get_errors(self):
         logger.debug("Item errors list is %s", errors)
         return errors
 
+    def set_skipped(self, answers, metadata):
+        # Process any conditional display rules
+        if self.schema_item:
+
+            if hasattr(self.schema_item, 'skip_condition') and self.schema_item.skip_condition:
+                rule = self.schema_item.skip_condition
+                all_conditions_met = True
+
+                for when in rule['when']:
+                    value = answers.get(when['id']) if 'id' in when else get_metadata_value(metadata, when['meta'])
+                    all_conditions_met &= evaluate_rule(when, value)
```

Comment:
Similar to above; no need to go through every `when` and OR these together; just break on the first False.